src/file-selector.ts

```diff
@@ -124,19 +124,26 @@ function fromDataTransferItem(item: DataTransferItem, entry?: FileSystemEntry |
     if (typeof (item as any).getAsFileSystemHandle === 'function') {
         return (item as any).getAsFileSystemHandle()
             .then(async (h: any) => {
+                if (!h) {
+                    return notAFile(item);
+                }
                 const file = await h.getFile();
                 file.handle = h;
                 return toFileWithPath(file);
             });
     }
     const file = item.getAsFile();
     if (!file) {
-        return Promise.reject(`${item} is not a File`);
+        return notAFile(item);
     }
     const fwp = toFileWithPath(file, entry?.fullPath ?? undefined);
     return Promise.resolve(fwp);
```

Comment:
It seem's that now the code will throw and exit with `not a file` and won't actually read the file when there is actually a valid file using the `item.getAsFile()` code below.

Do you know what was the initial benefit of using this experimental `getAsFileSystemHandle` function ?
Can we simply drop it ?

It would be great to fallback to the original code

Reply:
Agreed, this should be a fallback, not an exit case. We should also be adding some types here to ensure correctness.

Reply:
@buenjybar could you share more about how you get into that situation? Because if `getAsFileSystemHandle` exists and you're actually dropping a file, it should work in theory.

Reply:
Good point, that should work actually. The only time it would not is if you drop a directory I guess? But then also `.getAsFile()` would also not work I imagine.

Reply:
Should we go ahead with this change? As far as [getAsFileSystemHandle](https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/getAsFileSystemHandle) goes, we're doing the right thing, guarding for `null`.

Alternatively, we could make it go to L135 (on the right).

Reply:
Right now, when the handle is undefined/null, we reject with `not a file`. I'm not sure if that's what we want and how we even get there since we filter for files way before the function is called. I wasn't able to replicate outside cypress, but I would like to understand how it happens.

Alternatively, we could make the code path go to `getAsFile` when the handle is undefined/null?